frontend/src/scenes/trends/trendsLogic.test.ts

```diff
@@ -0,0 +1,164 @@
+import { BuiltLogic } from 'kea'
+import { mockAPI } from 'lib/api.mock'
+import { expectLogic, initKeaTestLogic } from '~/test/kea-test-utils'
+import { trendsLogic } from 'scenes/trends/trendsLogic'
+import { trendsLogicType } from 'scenes/trends/trendsLogicType'
+import { PropertyOperator, TrendResult } from '~/types'
+
+jest.mock('lib/api')
+
+describe('trendsLogic', () => {
+    let logic: BuiltLogic<trendsLogicType>
+
+    mockAPI(async ({ pathname, searchParams }) => {
+        if (pathname === '_preflight/') {
+            return { is_clickhouse_enabled: true }
+        } else if (pathname === 'api/users/@me/') {
+            return { organization: {}, team: { ingested_event: true, completed_snippet_onboarding: true } }
+        } else if (
+            [
+                'api/action/',
+                'api/projects/@current/event_definitions/',
+                'api/users/@me/',
+                'api/dashboard',
+                'api/insight',
+            ].includes(pathname)
+        ) {
+            return { results: [] }
+        } else if (['api/insight/session/', 'api/insight/trend/'].includes(pathname)) {
+            return { result: ['result from api'] }
+        } else {
+            throw new Error(`Unmocked fetch to: ${pathname} with params: ${JSON.stringify(searchParams)}`)
+        }
+    })
+
+    describe('core assumptions', () => {
+        initKeaTestLogic({
+            logic: trendsLogic,
+            props: {},
+            onLogic: (l) => (logic = l),
+        })
+
+        it('loads results on mount', async () => {
+            await expectLogic(logic).toDispatchActions(['loadResults'])
+        })
+    })
+
+    describe('reducers', () => {
+        initKeaTestLogic({
+            logic: trendsLogic,
+            props: {},
+            onLogic: (l) => (logic = l),
+        })
+
+        it('visibilityMap', async () => {
```

Comment:
On a side note, might have to choose a pattern to keep between visibilityMap in trendsLogic and hiddenMap in funnelLogic.

Reply:
Absolutely. Can you make an issue about it? The difference here is as well that visibilityMap is not persisted in the URL either.

Reply:
Yup! linking it here #6102